src/kafka-util/src/addr.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::error::Error;
+use std::fmt;
+use std::num::ParseIntError;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+/// Represents the address of a Kafka broker.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct KafkaAddr {
+    host: String,
+    port: u16,
+}
+
+impl FromStr for KafkaAddr {
+    type Err = KafkaAddrParseError;
+
+    fn from_str(s: &str) -> Result<KafkaAddr, Self::Err> {
+        let mut parts = s.splitn(2, ':');
+        let host = parts.next().unwrap();
```

Comment:
will this panic if s is the empty string?

Reply:
looks like no - it just gives the empty string. neat! i think expect is nice here still

Reply:
👍 done